### PR TITLE
[do not merge] HunyuanDIT refactor 

### PR DESCRIPTION
**Pull request by yiyixuxu**
part of https://github.com/huggingface/diffusers/pull/8240

to-do:

* [x] refactor `HunyuanDiT2DModel`: removed `HunyuanDiTPatchEmbed`, `HunyuanDiTTimestepEmbedder`
* [x]  refactor `apply_rotary_emb`
* [x] refactor `HunyuanDiTBlock` : keep it as a Hunyuan-specific block and move to the same file as `HunyuanDiT2DModel`

### CODE REVIEW

File: src/diffusers/models/attention.py [hunk 1]
```diff
@@ -84,214 +84,6 @@ def forward(self, x: torch.Tensor, objs: torch.Tensor) -> torch.Tensor:
...
-                            self.eps).to(origin_dtype)
-
-
-class FP32_SiLU(nn.SiLU):
-    def forward(self, inputs: torch.Tensor) -> torch.Tensor:
-        return torch.nn.functional.silu(inputs.float(), inplace=False).to(inputs.dtype)
-
-from typing import Tuple, Union, Optional
-
-class HunyuanDiTAttentionPool(nn.Module):
```
Comment (yiyixuxu) [inline src/diffusers/models/attention.py]: moved to hunyuan_transformer_2d.py

File: src/diffusers/models/transformers/hunyuan_transformer_2d.py [hunk 2]
```diff
@@ -12,129 +12,237 @@
...
+class FP32_Layernorm(nn.LayerNorm):
+    def forward(self, inputs: torch.Tensor) -> torch.Tensor:
+        origin_dtype = inputs.dtype
+        return F.layer_norm(inputs.float(), self.normalized_shape, self.weight.float(), self.bias.float(),
+                            self.eps).to(origin_dtype)
 
-    Image to Patch Embedding using Conv2d
 
-    A convolution based approach to patchifying a 2D image w/ embedding projection.
+class AdaLayerNormShift(nn.Module):
```
Comment (yiyixuxu) [inline src/diffusers/models/transformers/hunyuan_transformer_2d.py:44]: This is refactored from `default_modulation`: can you tell us more about it? Is it part of the `AdaLayerNorm` family but without the scale?  is there a reference? should we move to `embeddings.py` with other normalization methods?

File: src/diffusers/models/transformers/hunyuan_transformer_2d.py [hunk 3]
```diff
@@ -0,0 +1,511 @@
...
+        self.pooler = HunyuanDiTAttentionPool(self.text_len_t5, self.text_states_dim_t5, num_heads=8, output_dim=1024)
+
+        # Here we use a default learned embedder layer for future extension.
+        self.style_embedder = nn.Embedding(1, hidden_size)
+
+        # Image size and crop size conditions
+        self.extra_in_dim = 256 * 6 + hidden_size
+
+        # Text embedding for `add`
+        self.pos_embed = PatchEmbed(
```
Comment (yiyixuxu) [inline src/diffusers/models/transformers/hunyuan_transformer_2d.py:324]: I refactored this with our `PatchEmbed` but we are not actually adding the position embedding here - does it make sense to call it `pos_embed` still?
